cpp/test/core/unit_test.bond

```diff
@@ -664,6 +664,14 @@ struct WithSimpleList<T>
     1: required_optional simple_list<T> field;
 };
 
+using ValueWrapper<T> = T;
+
+struct EnumValueWrapper
```

Comment:
There's no code making any assertions about this struct, so all we're doing is testing that it compiles. We should also briefly test that is serializes/deserializes correctly.